backend/core/routes/chat_routes.py

```diff
@@ -266,9 +267,9 @@ async def create_stream_question_handler(
 
         gpt_answer_generator = OpenAIBrainPicking(
             chat_id=str(chat_id),
-            model=chat_question.model,
-            max_tokens=chat_question.max_tokens,
-            temperature=chat_question.temperature,
+            model=(brain_details or chat_question).model if current_user.user_openai_api_key else "gpt-3.5-turbo",
```

Comment:
@StanGirard the reason I put this is that there is currently a bug in the frontend where you can enter an API key and select GPT4 for a brain, but then come back and delete the API key and hit enter. It keeps GPT4 as the selected model, even though the dropdown changes, and then the host (you) pay for the GPT4 tokens. I think it's best practice to safeguard for this sort of thing in the backend and not rely on the frontend, but if you still want me to remove this I will. The number of users that take advantage of this may be small. If you want me to demo this bug I can attach a video or I can jump on a zoom call.

Reply:
You are amazing to think of that ! I agree then